doc/source/reference/demeshening.rst

```diff
@@ -0,0 +1,146 @@
+
+At the end of this process, each file will be associated with a single "coarse"
+index (which covers the entire domain of the data), as well as a set of
+"collision" locations, and in each "collision" location a set of bitarrays that
+correspond to that subregion.
+
+When reading data, yt will identify which "coarse" index regions are necessary
+to read.  If any of those coarse index regions are covered by more than one
+file, it will examine the "fine" index for those regions and see if it is able
+to subset more efficiently.  Because all of these operations can be done with
+logical operations, this considerably reduces the amount of data that needs to
+be read from disk before expensive selection operations are conducted.
+
+For those situations that involve particles with regions of influence -- such
+as smoothed particle hydrodynamics, where particles have associated smoothing
+lenghts -- these are taken into account when conducting the indexing system.
+
+Efficiency of Index Orders
+--------------------------
+
+What this can lead to, however, is situations where (particularly at the edges
+of regions populated by SPH particles) the indexing system identifies
+collisions, but the relatively small number of particles and correspondingly
+large "smoothing lengths" result in a large number of "fine" index values that
+need to be set.
+
+Counterintuitively, this actually means that occasionally the "fine" indexing
+process can take an inordinately long amount of time for *small* datasets,
+rather than large datasets.
+
+In these situations, it is typically sufficient to set the "fine" index order
+to be much lower than its default value.  For instance, setting the
+``index_order`` to (5, 3) means that the full domain will be subdivided into 32
+bins in each dimension, and any "collision" zones will be further subdivided
+into 8 bins in each dimension (corresponding to an effective 256 bins across
+the full domain).
+
+If you are experiencing very long index times, this may be a productive
+parameter to modify.
+
+Index Caching
+-------------
+
+The index values are cached between instantiation, in a sidecar file named with
+the name of the dataset file and the suffix ``.indexII_JJ.ewah``, where ``II``
+and ``JJ`` are ``index_order1`` and ``index_order2``.  So for instance, if
+``index_order`` is set to (5, 7), and you are loading a dataset file named
+"snapshot_200.hdf5", after indexing, you will have an index sidecar file named
+``snapshot_200.hdf5.index5_7.ewah``.  On subsequent loads, this index file will
+be reused, rather than re-generated.
+
+The filename scheme can be changed by supplying an alternate filename to the
+``load`` command with the argument ``index_filename``.  For instance, if you
+are accessing data in a read-only location, you can specify that the index will
+be cached in a location that is write-accessible to you.
+
+These files contain the *compressed* bitmap index values, along with some
```

Comment:
Can you give an estimate of the resulting file size ? I suppose it grows as `2^index_order1 + 2^index_order2` ?

Reply:
It doesn't, unfortunately -- it's compressed, so very dependent on the specific layout of the file.  In general it will be small, but I will expand this with information about this, repeating my comment.